99robots-header-footer-code-manager.php

```diff
@@ -1001,6 +1016,421 @@ public static function hfcm_request_handler()
             }
         }
 
+       
+        public static function hfcm_request_posts() {
+
+            // check user capabilities.
+            $nnr_hfcm_can_edit = current_user_can( 'manage_options' );
+
+            if ( !$nnr_hfcm_can_edit ) {
+                echo 'Sorry, you do not have access to this page.';
+                return false;
+            }
+
+            if ( isset( $_POST['insert'] ) ) {
+                // Check nonce
+                check_admin_referer( 'create-snippet' );
+            } else {
+                if ( empty( $_REQUEST['id'] ) ) {
+                    die( 'Missing ID parameter.' );
+                }
+                $id = absint( $_REQUEST['id'] );
+            }
+        
+            if ( isset( $_POST['insert'] ) ) {
+                // Check nonce
+                check_admin_referer( 'create-snippet' );
+            } else {
+                if ( ! isset( $_REQUEST['id'] ) ) {
+                    die( 'Missing ID parameter.' );
+                }
+                $id = (int) $_REQUEST['id'];
+            }
+            if ( isset( $_POST['update'] ) ) {
+                // Check nonce
+                check_admin_referer( 'update-snippet_' . $id );
+            }
+        
+            // Handle AJAX on/off toggle for snippets
+            if ( isset( $_POST['getPosts'] ) ) {
+        
+        
+                // Check nonce
+                check_ajax_referer( 'hfcm-get-posts', 'security' );
+        
+                // Global vars
+                global $wpdb;
+                $table_name = $wpdb->prefix . 'hfcm_scripts';
+        
+                // Get all selected posts
+                if ( -1 === $id ) {
+                    $s_posts = array();
+                    $ex_posts = array();
+                } else {
+        
+                    // Select value to update.
+                    $script = $wpdb->get_results( $wpdb->prepare( "SELECT s_posts from $table_name where script_id=%s", $id ) );
```

Comment:
clean this up, name should be nnr_hfcm_script so that variables are unique and follows standard naming convention